### PR TITLE
Gutted user configuration.

### PR DESCRIPTION
**Pull request by taphu**
See commits for comments.

### CODE REVIEW

File: src/Web/Moonshine.hs [hunk 1]
```diff
@@ -49,24 +49,24 @@ import qualified System.Metrics.Gauge as G (inc, Gauge, dec, set, add, subtract)
 {- |
   The moonshine monad.
 -}
-data Moonshine config a =
-  Moonshine (State config -> IO (State config, a))
+data Moonshine a =
+  Moonshine (State -> IO (State , a))
 
-instance Monad (Moonshine config) where
+instance Monad (Moonshine ) where
```
Comment (potomak) [inline src/Web/Moonshine.hs]: Are parenthesis needed anyway here?

Comment (taphu) [inline src/Web/Moonshine.hs]: Nope, I'll fix.

Comment (taphu) [inline src/Web/Moonshine.hs]: fixed.

**Comment by potomak** [comment]
Let me rephrase this: you're removing user config from the `Moonshine` type because services using moonshine can handle user config by using the canteven-config library, right?

**Comment by potomak** [comment]
Also I didn't understand well how *moonshine is exposing `canteven`*.
Can you explain it in another way please?

**Comment by MHova** [comment]
How does Moonshine "bring together a bunch of handy technologies" in this case in reference to `cant-even`? Does this version of Moonshine automatically make use of `cant-even`? Or will the client have to integrate that themselves?

(pretty much the same question as Giovanni just asked)

**Comment by potomak** [comment]
Oh ok, got it: without including a user config in the `Moonshine` type anymore, moonshine gives the *opportunity* to services to use cantevent-config by theirself.

**Comment by potomak** [comment]
...by using `getUserConfig`.

**Comment by MHova** [comment]
Okay I see :+1:

**Comment by taphu** [comment]
ok, good.. I was struggling with writing a longer explanation. :-)

**Comment by potomak** [comment]
canteven-config and moonshine are sharing the same `LoggingConfig` data type, could we make something in this PR about it or it's out of the scope?

**Comment by taphu** [comment]
Let's leave that out of scope for now. I will follow up with some improvements to both canteven-config and moonshine related to this.

**Comment by potomak** [comment]
:+1:

**Comment by taphu** [comment]
Thanks.
